operator/api/v1beta1/lokistack_types.go

```diff
@@ -343,9 +343,30 @@ type ObjectStorageSecretSpec struct {
 	Name string `json:"name"`
 }
 
+// ObjectStorageSchemaVersion defines the storage schema version which will be used with the Loki cluster.
+//
+// +kubebuilder:validation:Enum=v11;v12
+type ObjectStorageSchemaVersion string
+
+const (
+	// ObjectStorageSchemaV11 when using v11 for the storage schema
+	ObjectStorageSchemaV11 ObjectStorageSchemaVersion = "v11"
+
+	// ObjectStorageSchemaV12 when using v12 for the storage schema
+	ObjectStorageSchemaV12 ObjectStorageSchemaVersion = "v12"
+)
+
 // ObjectStorageSpec defines the requirements to access the object
 // storage bucket to persist logs by the ingester component.
 type ObjectStorageSpec struct {
+	// Type of object storage that should be used
```

Comment:
Doc requires update explaining schema version. In addition we should highlight here that using/updating this field is an append only operation, means we support only future log chunks being stored in the new schema version. Further on we need to give here a thorough explanation of what `future log chunks` means, e.g. 24h after successful reconciliation of the schema version on all components.